indra/newview/gltfscenemanager.cpp

```diff
@@ -126,6 +133,170 @@ void GLTFSceneManager::decomposeSelection()
     }
 }
 
+void GLTFSceneManager::uploadSelection()
+{
+    if (mUploadingAsset)
+    { // upload already in progress
+        LLNotificationsUtil::add("GLTFUploadInProgress");
+        return;
+    }
+
+    LLViewerObject* obj = LLSelectMgr::instance().getSelection()->getFirstRootObject();
+    if (obj && obj->mGLTFAsset)
+    {
+        // make a copy of the asset prior to uploading
+        mUploadingAsset = std::make_shared<Asset>();
+        mUploadingObject = obj;
+        *mUploadingAsset = *obj->mGLTFAsset;
+
+        GLTF::Asset& asset = *mUploadingAsset;
+
+        for (auto& image : asset.mImages)
+        {
+            if (!image.mData.empty())
+            {
+                mPendingImageUploads++;
+
+                LLPointer<LLImageRaw> raw = new LLImageRaw(image.mWidth, image.mHeight, image.mComponent);
+                U8* data = raw->allocateData();
+                llassert_always(image.mData.size() == raw->getDataSize());
+                memcpy(data, image.mData.data(), image.mData.size());
+
+                // for GLTF native content, store image in GLTF orientation
+                raw->verticalFlip();
```

Comment:
We should store images from GLTF models in the same orientation as other image content, so that they are compatible with GLTF material assets. At the same time, we want to store GLTF models in their original state as well. Therefore, the appropriate solution would be to flip the UVs at runtime in the shader. It's quite possible that the given texture transform functions in the shader are not sufficient for handling the GLTF model case.